ideascube/tests/test_views.py

```diff
@@ -511,3 +517,55 @@ def test_by_tag_page_is_paginated(app, monkeypatch):
     assert not response.pyquery.find('.next')
     assert response.pyquery.find('.previous')
     response = app.get(url + '&page=3', status=404)
+
+
+def test_build_package_card_info_must_no_fail_for_no_package(app, systemuser):
+    from ideascube.configuration import set_config
+    from ideascube.views import build_package_card_info
+    set_config('home-page', 'displayed-package-ids', ['test.package1'], systemuser)
+    assert build_package_card_info() == []
+
+
+def test_build_package_card_info(app, systemuser, catalog):
+    from ideascube.configuration import set_config
+    from ideascube.views import build_package_card_info
+    catalog.create_package(id = 'test.package1.fr',
+                           name='Test package1',
+                           description='Test package1 description',
+                           template_id='template_name',
+                           language='fr',
+                           staff_only=False)
+    catalog.create_package(id='test.package2',
+                           name='Test package2',
+                           description='Test package2 description',
+                           template_id='template_name',
+                           language='fr',
+                           staff_only=False)
+    set_config('home-page', 'displayed-package-ids', ['test.package1.fr'], systemuser)
+
+    # The main information here is that the card id is the package id minus
+    # the last .fr
+    with catalog:
```

Comment:
The `create_package` is to configure the catalog mocker. It's not important if you configure the mock inside the context or not.

You just have to add the package to the mock before your code **use** the (mocked) catalog.
And the context manager is just here to mock the real underlying catalog API.